### DISCUSSION TITLE
Service to add/remove/update zones

### DISCUSSION THREAD

**Discussion by martinhoess** [frenck/spook#155] [Ideas] [answered]
HA itself currently has no possibility to dynamically create/modify/delete zones (lat/long/radius).

However, this function would be quite practical for a few applications.

https://community.home-assistant.io/t/device-tracer-and-dynamic-zones/567829
https://community.home-assistant.io/t/create-a-zone-around-a-moving-person/473978
Another use case would be, for example, to operate HA in a camper van and make the current location automatically the home zone.

I have no idea how the zones in HA are structured internally and if this is even currently possible.

**Comment by vingerha** [comment]
+1

**Comment by frenck** [comment]
I've converted the issue into a discussion (as it is not a bug at this point).

Anyways, I like that idea @martinhoess 👍 
I'll keep it in mind for an future iteration (or maybe someone beats me to it).

../Frenck

**Comment by frenck** [comment]
Picking this one up.

**Comment by frenck** [comment]
![CleanShot 2023-08-11 at 14 35 11](https://github.com/frenck/spook/assets/195327/72841993-2b88-4f13-880b-dcb27fe57e66)


![CleanShot 2023-08-11 at 14 35 39](https://github.com/frenck/spook/assets/195327/0083b93b-6b5a-4503-8bd6-b9c57a2110c0)

**Reply by frenck** [reply, marked as answer]
Implemented `zone.create`, `zone.update` and `zone.delete` in: <https://github.com/frenck/spook/pull/304>

**Reply by frenck** [reply]
Shipped in [v1.1.0](https://github.com/frenck/spook/releases/tag/v1.1.0).

**Reply by martinhoess** [reply]
Wow, that was fast.
Now I just have to remember what I originally was making that I would have needed this for. 😄

THANKS!

**Reply by vingerha** [reply]
My use case is to create a zone when the person is 'stable' in a specific place and then the zone is used to measure entry/exit. Stable is that the lat/lon are somewhat the same.

**Reply by frenck** [reply]
Nice, that is actually pretty genius 👍

**Reply by vingerha** [reply]
The added perk is that the zone is usable in the history cards, being a state... where lat/lon is not (really)

**Comment by martinhoess** [comment]
Unfortunately this does not work for me

![image](https://github.com/frenck/spook/assets/47973549/8d182423-e747-44f5-9d34-f1d630c2b504)

**Reply by frenck** [reply]
Can you please open up an issue? Thanks 👍
